test/unit/gateways/barclaycard_smartpay_test.rb

```diff
@@ -41,6 +86,48 @@ def test_successful_purchase
     assert response.test?
   end
 
+  def test_successful_purchase_with_alternate_address
+    response = stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options_with_alternate_address)
+    end.respond_with(successful_authorize_response, successful_capture_response)
```

Comment:
It might be good to either `.check_request` the fallback parsing of the `address1: '新北市店溪路3579號139樓'` here, or just remove the test, since there's not much point in testing the success of the mock transaction.

Reply:
👍 Good point -- `.check_request` now added.The fallback parsing will return 'Not Provided' for the street address which is a new addition.